opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/EventConfRestService.java

```diff
@@ -449,6 +450,90 @@ public Response getEventsByVendor(final String vendorName, SecurityContext secur
+                LOG.warn("Attempt to create duplicate EventConfSource. name='{}'", request.getName());
+                return Response.status(Response.Status.CONFLICT)
+                        .entity("An EventConfSource with the same name already exists")
+                        .build();
+            }
+
+            final String username = getUsername(securityContext);
+
+            int maxFileOrder = Optional.ofNullable(eventConfSourceDao.findMaxFileOrder()).orElse(0);
+            int newOrder = maxFileOrder + 1;
+
+            LOG.debug("Assigned fileOrder={} for new EventConfSource name='{}'",
+                    newOrder, request.getName());
+
+            EventConfSource eventConfSource = new EventConfSource();
+            eventConfSource.setName(request.getName());
+            eventConfSource.setDescription(request.getDescription());
+            eventConfSource.setVendor(request.getVendor());
+            eventConfSource.setEnabled(true);
+            eventConfSource.setEventCount(0);
+            eventConfSource.setCreatedTime(now);
+            eventConfSource.setLastModified(now);
+            eventConfSource.setFileOrder(newOrder);
+            eventConfSource.setUploadedBy(username);
+
+            LOG.debug("Persisting EventConfSource name='{}'", request.getName());
+
+            Long eventConfSourceId = eventConfPersistenceService.createEventConfSource(eventConfSource);
+
+            LOG.info("Successfully created EventConfSource id={}, name='{}', user='{}'",
+                    eventConfSourceId, request.getName(), username);
+
+            return Response.status(Response.Status.CREATED)
+                    .entity(Map.of(
+                            "id", eventConfSourceId,
+                            "name", eventConfSource.getName(),
+                            "fileOrder", eventConfSource.getFileOrder()
+                    ))
+                    .build();
+
+        } catch (Exception ex) {
+            LOG.error("Failed to create EventConfSource name='{}'. Error={}",
+                    request.getName(), ex.getMessage(), ex);
+
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity("Unexpected error occurred while creating EventConfSource: "
+                            + ex.getMessage())
+                    .build();
+        }
+    }
+
+    private void validateAddSourceRequest(final AddEventConfSourceRequest request) {
+        if (request == null) {
+            throw new IllegalArgumentException("Request must not be null.");
+        }
+
+        if (request.getName() == null || request.getName().isBlank()) {
+            throw new IllegalArgumentException("Source name must not be null or blank.");
+        }
+
+        if (request.getVendor() == null || request.getVendor().isBlank()) {
+            throw new IllegalArgumentException("Vendor must not be null or blank.");
+        }
```

Comment:
It looks like we enforce limits in the DB schema. May be we have to validate the lengths.